src/arithmetic/rewrite_simplify.cc

```diff
@@ -1025,20 +1068,45 @@ Mutate_(const LT* op, const Expr& self) {
     TVM_TRY_REWRITE_IF(x * c1 < y * c1, y < x,
                        c1.Eval()->value < 0);
 
-    // require c1 > 0 to work for any div mode
     TVM_TRY_REWRITE_IF(x * c2 < c1, x < (c1 - 1) / c2 + 1,
                        c1.Eval()->value > 0 &&
                        c2.Eval()->value > 0);
-    TVM_TRY_REWRITE_IF(x / c1 < c2, x < c1 * c2,
-                       c1.Eval()->value > 0 &&
+    TVM_TRY_REWRITE_IF(x * c2 < c1, x < c1 / c2,
+                       c1.Eval()->value <= 0 &&
                        c2.Eval()->value > 0);
```

Comment:
This is true if `x` is an integer. Can we assume that?

Reply:
I think we can, since these rules are guarded by the condition `IsIndexType(op->a.type())`, and also all constants are declared as integers (`PVar<Integer>`).